### ISSUE TITLE
vscode: Fails to register service workers breaking web views

### ISSUE THREAD

**Issue by jakehamilton** [NixOS/nixpkgs#188222]
### Describe the bug

Running VS Code results in an error any time it tries to render a web view. This breaks core
functionality such as release notes, extensions pages/panels.

### Steps To Reproduce

1. Run the following commands.

```bash
export NIX_ALLOW_UNFREE=1

nix run --impure nixpkgs#vscode
```
2. See any webview fail to load. This can be the release notes or any extension's page.

### Expected behavior

VS Code webviews work and can display their contents.

### Screenshots

<img width="1136" alt="vs code screenshot showing an empty page that failed to render and the error message below this image" src="https://user-images.githubusercontent.com/7005773/186522655-ae2332d7-b0bc-4035-9da8-f00ca49e17ed.png">

The error text:

```
Error loading webview: Error: Could not register service workers: InvalidStateError: Failed to register a ServiceWorker: The document is in an invalid state..
```

### Additional context

I have not noticed this problem on my NixOS systems, but it appears to be a problem on macOS.

### Notify maintainers

@eadwu @Synthetica9 @maxeaubrey @bobby285271

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-darwin"`
 - host os: `Darwin 21.6.0, macOS 10.16`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.8.1`
 - channels(jhamilton7): `"darwin, home-manager, nixpkgs"`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/etc/nix/inputs/nixpkgs`
```

**Comment by ck3d** [comment]
I can confirm this issue on darwin arm64 also for version 1.70.2 . But I have not seen this error for versions 1.69.2 and 1.70.0

**Comment by jakehamilton** [comment]
I should have put the version of VS Code in the issue, apologies for that. Here is what I get when I check the version (with my flake locked on `nixos/nixpkgs/release-22.05`):

```bash
➜ nix run .#vscode -- --version                          
1.69.2
3b889b090b5ad5793f524b5d1d39fda662b96a2a
x64
```

Switching my flake to `nixos/nixpkgs` so it'll be using `master`, I get:

```bash
➜ nix run .#vscode -- --version 
1.70.2
e4503b30fc78200f846c62cf8091b76ff5547662
x64
```

The issue persists for both of these versions unfortunately.

**Comment by jakehamilton** [comment]
I asked two coworkers that were still on Big Sur to try running VS Code with Nix and I got some interesting results:

## Coworker A

Reproduced the issue:

<img width="996" alt="image (1)" src="https://user-images.githubusercontent.com/7005773/187278379-1cd45f4d-108c-48ff-80b2-4391f14481c8.png">

<details>
  <summary>nix-info output</summary>
  
```
 - system: `"x86_64-darwin"`
 - host os: `Darwin 20.3.0, macOS 10.16`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.4`
 - channels(root): `"nixpkgs"`
 - channels(ckisielius): `"darwin"`
 - nixpkgs: `/etc/nix/inputs/nixpkgs`
```
  
</details>

## Coworker B:

Could not reproduce the issue:

<img width="1790" alt="Screen Shot 2022-08-29 at 11 50 34 AM" src="https://user-images.githubusercontent.com/7005773/187279035-e5d84fe5-d417-47d7-b327-bf474f1d9399.png">

<details>
  <summary>nix-info output</summary>
  
```
 - system: `"x86_64-darwin"`
 - host os: `Darwin 20.6.0, macOS 10.16`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.11.0`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

**Comment by jakehamilton** [comment]
I just went and tested again in case I missed something and it appears that the issue may have been running two separate instances of VS Code at the same time. I had always been executing my `nix run nixpkgs#vscode` from *within* the VS Code terminal. That would make the issue happen every time.

I tried quitting all VS Code instances and running `nix run nixpkgs#vscode` from iTerm2 and, perhaps unsurprisingly, it worked perfectly well!

I'll be closing this issue as the problem was due to running two different instances of VS Code and *not* about NixPkgs' build of VS Code.
